src/react/ShikiMagicMove.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react'
+import type { HighlighterCore } from 'shiki/core'
+import { codeToKeyedTokens, createMagicMoveMachine } from '../core'
+import type { MagicMoveDifferOptions, MagicMoveRenderOptions } from '../types'
+import { ShikiMagicMoveRenderer } from './ShikiMagicMoveRenderer'
+import { createCSSPropertiesFromString } from './utils'
+
+export function ShikiMagicMove(
+  {
+    highlighter,
+    lang,
+    theme,
+    code,
+    options,
+    onStart,
+    onEnd,
+    className,
+  }: {
+    highlighter: HighlighterCore
+    lang: string
+    theme: string
+    code: string
+    options?: MagicMoveRenderOptions & MagicMoveDifferOptions
+    onStart?: () => void
+    onEnd?: () => void
+    className?: string
+  },
+) {
+  const machine = React.useMemo(
+    () => createMagicMoveMachine(
+      code => codeToKeyedTokens(highlighter, code, {
+        lang,
+        theme,
+      }),
+      options,
+    ),
+    [highlighter, lang, options, theme],
```

Comment:
We shall not create new machines every time they change. It's also possible to use Magic Move to transit across themes and langs.

Reply:
This is because I need to fight with the rules of React hooks. I clean the dependency array for now, and if there are any issues afterward, I will try to fix them.